### ISSUE TITLE
Override generated query

### ISSUE THREAD

**Issue by callmekatootie** [aws-amplify/amplify-cli#8366]
**Which Category is your question related to?**

Graphql

**Amplify CLI Version** 

6.0.0

**What AWS Services are you utilizing?**

Auth, Functions, Graphql, Storage

**Provide additional details e.g. code snippets**

I have a schema like this:

```
type Foo @model {
  id: ID!
  .... other attributes
}
```

I understand that on amplify push, I automatically get the type Query, with `getFoo(id: ID)`. All well.

After a while, I wish to override this query - I want to use the `@function` directive to call a lambda function for this query only. How do I go about with it? When I tried to define it as:

```
type Query {
  getFoo(id: ID!): Foo @function(name: "blah")
}
```

I got the following error on `amplify push` :

```
Object type extension 'Query' cannot redeclare field getFoo
An error occurred during the push operation: Object type extension 'Query' cannot redeclare field getFoo
```

From what I see, I should rename the getFoo query to something else and then attach the function to it. I can no longer use `getFoo` itself... Is that right or there is a way to override?

**Comment by callmekatootie** [comment]
Thank you for your quick response. I was not expecting it so soon, so I ended up taking the following approach:

```
type Foo @model {
  id: ID!
}

type Query {
   getFooBar(id: ID!): Foo @function(name: "hello world")
}
```

Instead of `getFoo`, I defined a new query named `getFooBar`. And there was peace.

Now that I see your response, I tried your approach. I set `getFoo` to null in the type Foo definition and then in the query renamed `getFooBar` to `getFoo`.

But amplify didn't like it. The compile was a success but it appears that the cloudformation / appsync deployment failed:

```
Following resources failed

Resource Name: QuerygetFooResolver (AWS::AppSync::Resolver)
Event Type: create
Reason: Only one resolver is allowed per field.
(Service: AWSAppSync; Status Code: 400; Error Code: BadRequestException; Request ID: redacted; Proxy: null)
```

Perhaps it didn't delete the default `getFoo` resolver? So it thinks my function based resolver is a duplicate? Anyway, I have reverted my changes to the workaround mentioned earlier - will wait for your response nevertheless since having `getFoo` query is better than `getFooBar`...

**Comment by yuth** [comment]
@PatrykMilewski you are right. When you override a query, mutation or a subscription, you will have to specify the field names of all the operations that you are not overriding. So in this case the schema will look like if you want the list query to be generated.

```gql
type Foo @model(queries: { get: null, list: "listFoo" }) {
  id: ID!
}

```
